integration/test_web.py

```diff
@@ -0,0 +1,496 @@
+import sys
+import time
+import shutil
+import json
+import urllib2
+from os import mkdir, unlink, utime
+from os.path import join, exists, getmtime
+
+import allmydata.uri
+
+import util
+
+import requests
+import pytest_twisted
+import html5lib
+from bs4 import BeautifulSoup
+
+
+def test_index(alice):
+    """
+    we can download the index file
+    """
+    util.web_get(alice, u"")
+    # ...and json mode is valid json
+    json.loads(util.web_get(alice, u"", params={u"t": u"json"}))
+
+
+def test_upload_download(alice):
+    """
+    upload a file, then download it via readcap
+    """
+
+    FILE_CONTENTS = u"some contents"
+
+    readcap = util.web_post(
+        alice, u"uri",
+        data={
+            u"t": u"upload",
+            u"format": u"mdmf",
+        },
+        files={
+            u"file": FILE_CONTENTS,
+        },
+    )
+    readcap = readcap.strip()
+
+    data = util.web_get(
+        alice, u"uri",
+        params={
+            u"uri": readcap,
+            u"filename": u"boom",
+        }
+    )
+    assert data == FILE_CONTENTS
+
+
+def test_put(alice):
+    """
+    use PUT to create a file
+    """
+
+    FILE_CONTENTS = u"added via PUT"
+
+    resp = requests.put(
+        util.node_url(alice.node_dir, u"uri"),
+        files={
+            u"file": FILE_CONTENTS,
+        },
+    )
+    cap = allmydata.uri.from_string(resp.text.strip().encode('ascii'))
+    assert isinstance(cap, allmydata.uri.CHKFileURI)
+    assert cap.size == 153
+    assert cap.total_shares == 4
+    assert cap.needed_shares == 2
```

Comment:
Maybe these three magic numbers could be eliminated?

Reply:
"Yes, sure" I thought. And the needed/total is easy .. but why is the size of that 153 and not `len(FILE_CONTENTS)`? (Looks like everything is "140 + actual-size" .. but the URI Extension Block is not 140 bytes).

Reply:
Ahh, nice .. bug in the test :/ the 140 overhead is the mime headers; should have used `data=` not `files=` ...